learn/guides/ballerina-openapi-support.md

```diff
@@ -0,0 +1,314 @@
+```bash
+$ bal openapi -i <openapi contract>  [--tags < "tag1","tag2">]
+```
+- ##### - -operations
+To generate the Ballerina client or service stub with a subset of operations defined in the OpenAPI contract, use the options `--operations` and specify the operations you need as specified in the OpenAPI definition.
+```bash
+$ bal openapi -i <openapi contract> [--operations <"op1", "op2">]
+```
+- ##### - -license
+If you want to generate the Ballerina files with the given copyright or license header, you can use this `--license` flag with your copyright text.
+```bash
+$ bal openapi -i <openapi contract> [--license <license-file-path>]
+```
+- ##### -n |- -nullable
+This is an optional flag in the OpenAPI to Ballerina command. If your OpenAPI specification includes JSON schema properties that are not marked as **nullable:true**, they may return as null in some responses. It will result in a JSON schema to Ballerina record data binding error. If you suspect this can happen for any property, it is safe to generate all data types in the generated record with Ballerina nil support by turning on this flag.
+```bash
+$ bal openapi -i <openapi contract> [-n |--nullable]
+```
+- ##### - -with-tests
+This is optional. Work with the client generation command and generate a test boiler-plate for all the remote functions of the generated client.
+
+## OpenAPI Annotation Reference
+The `@openapi:ServiceInfo` annotation support for several usages in the Ballerina OpenAPI tools. It contains attributes `contract` , `tag`, `operations`, `failOnErrors`, `excludeTags`, `excludeOperations`, `title`, `version` and `embed` for each particular purpose. These attributes are optional in annotation.
+
+```ballerina
+@openapi:ServiceInfo{
+    [contract: “/path/to/openapi.json|yaml”],
+    [ tag : “store” ],
+    [ operations: [“op1”, “op2”] ], 
+    [ failOnErrors]: true/false → default : true,
+    [ excludeTags ]: [“pets”, “user”],
+    [ excludeOperations: [“op1”, “op2”] ],
+    [ title : “store” ],
+    [ ‘version: “0.1.0” ],
+    [embed: false/true -> default: true]
+   }
+service greet on new http:Listener(9090) {
+    ...
+}
+```
+- ##### Contract: string?
+Here, you can provide a path to the OpenAPI contract as a string and the OpenAPI file can either be `.yaml` or `.json`. 
+- ##### Tags: string[ ]?
+The compiler will only validate resources against operations, which are tagged with a tag specified in the list. If not specified, the compiler will validate resources against all the operations defined in the OpenAPI contract.
+- ##### Operations: string[ ]?
+Should contain a list of operation names that need to be validated against the resources in the service. If not specified, the compiler will validate resources against all the operations defined in the OpenAPI contract. If both tags and operations are defined, it will validate against the union set of the resources.
```

Comment:
```suggestion
This should contain a list of operation names that need to be validated against the resources in the service. If not specified, the compiler will validate resources against all the operations defined in the OpenAPI contract. If both tags and operations are defined, it will validate against the union set of the resources.
```